tests/test_rank_stats.py

```diff
@@ -55,4 +55,18 @@ def test_percentage_difference(self):
 
     def test_mean_reciprocal_rank(self):
         self.rank_stats.reciprocal_ranks = [0.2, 0.4, 0.5, 0.6, 0.8]
+        self.rank_stats.total = 5
         self.assertEqual(self.rank_stats.mean_reciprocal_rank(), 0.5)
+
+    def test_mean_reciprocal_rank_missing_cases(self):
+        self.rank_stats.reciprocal_ranks = [0.2, 0.4, 0.5, 0.6, 0.8]
+        self.rank_stats.total = 20
+        self.assertEqual(self.rank_stats.mean_reciprocal_rank(), 0.1)
+
+    def test_return_mean_reciprocal_rank(self):
+        self.rank_stats.reciprocal_ranks = [0.2, 0.4, 0.5, 0.6, 0.8]
+        self.assertEqual(self.rank_stats.return_mean_reciprocal_rank(), 0.5)
+
+    def test_return_mean_reciprocal_rank_from_mrr_variable(self):
+        self.rank_stats.mrr = 0.1
+        self.assertEqual(self.rank_stats.return_mean_reciprocal_rank(), 0.1)
```

Comment:
It would return 0.1. There wouldn't be an instance where they would both be assigned. The reciprocal_ranks are assigned through the whole benchmark command while the mrr is assigned only when calling the generate plots command.

Reply:
As discussed, this is sub-par and would be clearer using some kind of immutable data class but that's not going to really roll here thanks to language constraints.